NSS-Enthusiastic-Camels/BangazonAPI
# Order resources

## Link to Ticket
closes # [Link](https://github.com/NSS-Enthusiastic-Camels/BangazonAPI/issues/5)

## Description of Proposed Changes

- Create model for Order
- Created mvc for Order


## Steps to Test



Outline the steps to test


```
1. git fetch
git checkout orderResources

2.Using the SQL file provided it will insert dummy data into the database

Test Get all:
in postman get with URL
```
http://localhost:5000/api/order
```
4.
Test Get single:
in postman get with URL
```
http://localhost:5000/api/order/1
```
5.
Test Post:
in postman post with URL
```
http://localhost:5000/api/order
```
in body paste
```
    {
        "CustomerId": 1
    }
```
Get departments again to see outcome.

6.
Test Put:
in postman put with URL
```
http://localhost:5000/api/order/4
```
in body paste
```
    {
        "paymentTypeId": 1
    }
```
Get departments to see outcome.

7.
Test ?_include=products for all orders:
in postman get all with URL
```
http://localhost:5000/api/order?_include=products
```
It should show all products in all orders.

Test ?_include=products for single order:
in postman get all with URL
```
http://localhost:5000/api/order/1?_include=products
```
It should show all products in the order with the id of 1. To test any other order, just place the appropriate order id before the "?".

8.
Test ?_include=customers for all orders:
in postman get with URL
```
http://localhost:5000/api/order?_include=customers
```
It should show the customer first and last name for all orders.

test ?_include=customers for single orders:
in postman get with URL
```
http://localhost:5000/api/order/1?_include=customers
```
It should show the customer first and last name for the order with id 1. To test any other order, just place the appropriate order id before the "?".
9.
Test completed for all orders:
in postman get with URL
```
http://localhost:5000/api/order?completed=true
```
It should show all of the orders that contain a paymentTypeId that are not null.
```


## Impacted Areas in Application

List general components of the application that this PR will affect:

* OrderModel and OrderController


## Mentions @username

@alexfont321
@stjepad
@priyankagargc27


## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
